institutions_api/db/db.py

```diff
@@ -161,8 +169,39 @@ def _update_institution_ror_id(session: Session, institution: Institution, ror_i
         # create a new ROR ID for the institution
         session.add(InstitutionIdentifier(ror_id_type, ror_id, institution.id))
 
+def _update_institution_unit_id(session: Session, institution: Institution, unit_id: str):
+    """ Handle updates to an institution's joined InstitutionIdentifier of type 'unitid'
+    based on the 'unitid' value passed in the API model
+    """
+
+    unit_id_type = _unit_id_type(session)
+
+    if not unit_id_type:
+        raise HTTPException(400, "IdentifierType for 'unitid' not found")
+
+    if not unit_id:
+        # delete any existing unit ids if null in the text fields
+        session.delete(institution.ipeds_metadata)
+        session.execute(delete(InstitutionIdentifier)
+            .where(InstitutionIdentifier.institution_id == institution.id)
+            .where(InstitutionIdentifier.identifier_type_id == unit_id_type.id))
+    else:
+        # Check if the institution already has a unitid
+        existing_unitid = [i for i in institution.identifiers if i.identifier_type_id == unit_id_type.id]
+
+        if existing_unitid:
+            # Update the existing unitid
+            existing_unitid[0].identifier = unit_id
+            session.add(existing_unitid[0])
+        else:
+            # Create a new InstitutionIdentifier for unitid if it doesn't exist
+            new_unitid = InstitutionIdentifier(identifier_type=unit_id_type, identifier=unit_id,
```

Comment:
Both parts of this block should handle updating the IPEDS_METADATA as well. 

The top should modify a existing record with the new data. 

The bottom should create a new row of IPEDS data and link it to the new id.